make-maas.sh

```diff
@@ -51,7 +51,26 @@ then
     echo "To install the backport: apt install -t xenial-backports lxc lxc-client"
 else
     echo "Setting up pxe redirect for IP $IPADDRESS"
-    lxc network set lxdbr0 raw.dnsmasq dhcp-boot=pxelinux.0,$CONTAINER,$IPADDRESS && systemctl restart lxd.service
+    if lxc network set lxdbr0 raw.dnsmasq dhcp-boot=pxelinux.0,$CONTAINER,$IPADDRESS
+    then
+        if snap services | grep -e '^lxd.daemon\s\+enabled\s\+active$' > /dev/null 2>&1
+        then
+            # On Cosmic, "systemctl restart lxd.service" doesn't work since LXD is now a snap.
+            # My best guess is that "snap restart lxd.daemon" is the proper way to do this now?
+            # Please correct if there is something better...
+            #
+            # Also, I'm curious - why is the restart necessary?
+            #
+            # UPDATE: Doing the restart at this particular timing seems to interfere with startup
+            # of the container.  This makes sense, but it again makes me wonder *why* we restart
+            # in some cases; there likely was some reason.  Anyway, for now, making this a no-op.
+            #sudo snap restart lxd.daemon
+            true  # do nothing for now
```

Comment:
So if I recall, and it has been a while the restart was required for the bridge change to take affect. I also don't remeber in the pre-snap days it affecting the container. Having said that, I haven't actually tested this with a snap so I'll look to your advice on what's working for you. 

A quick look through some of the LXD and snapcraft forums found this guidance from Stephan (LXD lead) "systemctl reload snap.lxd.daemon is in general safe to do (will not affect running containers)"

So I'd say for snaps a daemon reload is what we're really looking for in the new-era of snap based LXD.

Reply:
Got it.

I tried the systemctl reload, and it seems to work without problems.  I'm not 100% it's necessary, but it certainly doesn't seem to hurt, so maybe that's most in line with the rest of the current code.  Patch pushed.